server/src/main/java/org/elasticsearch/action/admin/cluster/stats/TransportClusterStatsAction.java

```diff
@@ -141,7 +141,9 @@ public TransportClusterStatsAction(
         this.analysisStatsCache = new MetadataStatsCache<>(threadPool.getThreadContext(), AnalysisStats::of);
         this.remoteClusterService = transportService.getRemoteClusterService();
         this.settings = settings;
-        this.remoteClusterStatsAction = remoteClusterStatsAction;
+
+        // register remote-cluster action with transport service only
```

Comment:
NIT: Sorry for this (and I admit that this is partly me being stupid) but maybe add "and not with the client" or so to make it clear what this is about. At least for me personally this was quite confusing and until today I have not once thought about the effects in the `Client` I believe :D